### DISCUSSION TITLE
activationEvents is not required anymore in vscode , and theia doesnot work if i do the same change in theia

### DISCUSSION THREAD

**Discussion by nive20** [eclipse-theia/theia#12370] [General]
The vscode 1.75.0 and above generates activation events automatically and we don't have to explicitly specify them as we do now and this is suggested from Microsoft as this impacts performance. We need to remove and support this in theia too if this is supported.
Is there any plan to include this vscode functionality in theia as well ?

**Comment by msujew** [comment]
Hey @nive20, this feature has already been requested in https://github.com/eclipse-theia/theia/issues/12161 and addressed in https://github.com/eclipse-theia/theia/pull/12167. Version `1.35.0` of Theia has been released with this feature included. Does something not work as expected with it?

**Reply by nive20** [reply]
@msujew  Ya i have tried this in 1.35 and this still doesn't work , I checked the same vsix file in vscode , there it works , but when i check the same changes in theia , it doen't work.
VSCODE version : 1.76.0
Theia version : 1.35.0

**Reply by vince-fugnitto** [reply]
@nive20 which extension have you tried which did not work correctly?
I re-verified using a simple [hello-world](https://github.com/eclipse-theia/theia/files/11158332/hello-world-0.0.1.zip) extension which does not declare any `activationEvents` and it works as expected.
